PyIlmBase/CMakeLists.txt

```diff
@@ -77,25 +77,23 @@ find_package(Boost OPTIONAL_COMPONENTS
   ${PYILMBASE_BOOST_PY2_COMPONENT}
   python3
   ${PYILMBASE_BOOST_PY3_COMPONENT})
-if(Boost_python_FOUND AND NOT(
-    Boost_python2_FOUND OR
-    Boost_${PYILMBASE_BOOST_PY2_COMPONENT}_FOUND OR
-    Boost_python3_FOUND OR
-    Boost_${PYILMBASE_BOOST_PY3_COMPONENT}_FOUND))
-  # old boost case, I guess we just warn and assume it is python2 (likely)
-  message(WARNING "Ambiguous boost python module found, assuming python 2")
-  set(PYILMBASE_BOOST_PY2_COMPONENT python)
-  set(PYILMBASE_BOOST_PY3_COMPONENT pythonIgnore)
-else()
-  if(Boost_python2_FOUND AND NOT Boost_${PYILMBASE_BOOST_PY2_COMPONENT}_FOUND)
-    message(WARNING "Legacy Boost python2 found, but does not include minor version, this is an old configuration and may not be portable")
-    set(PYILMBASE_BOOST_PY2_COMPONENT python2)
-  endif()
-  if(Boost_python3_FOUND AND NOT Boost_${PYILMBASE_BOOST_PY3_COMPONENT}_FOUND)
-    message(WARNING "Legacy Boost python3 found, but does not include minor version, this is an old configuration and may not be portable")
-    set(PYILMBASE_BOOST_PY3_COMPONENT python3)
-  endif()
-  if(NOT Boost_python_FOUND AND NOT(Boost_python2_FOUND OR Boost_python3_FOUND))
+ 
```

Comment:
Yeah, I think what I did adds another failure case that wasn't being handled before, so I think I prefer my change, but glad you found my logic error, thanks!

Reply:
Yes I saw you are handling more cases. Just included this temporarily to get a build including PyIlmBase. I'll remove it from this PR once you've merged your changes.

Reply:
I've merged in my changes. Hopefully they also fix some of the issues with the windows side as well...